### PR TITLE
Pin dependencies for the 0.5 release branch

### PR DESCRIPTION
**Pull request by lilic**
cc @coreos/team-monitoring

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -89,11 +89,12 @@ $ minikube addons disable metrics-server
...
 ### Kubernetes compatibility matrix
 
-| kube-prometheus stack | Kubernetes 1.14 | Kubernetes 1.15 | Kubernetes 1.16 | Kubernetes 1.17 |
-|-----------------------|-----------------|-----------------|-----------------|-----------------|
+| kube-prometheus stack | Kubernetes 1.14 | Kubernetes 1.15 | Kubernetes 1.16 | Kubernetes 1.17 | Kubernetes 1.18 |
+|-----------------------|-----------------|-----------------|-----------------|-----------------|-----------------|
 | `release-0.3`         | ✔              | ✔              | ✔              | ✔              |
 | `release-0.4`         | ✗              | ✗              | ✔              | ✔              |
-| `HEAD`                | ✗              | ✗              | ✗              | ✔              |
+| `release-0.5`         | ✗              | ✗              | ✗              | ✔              | ✔
```
Comment (brancz) [inline README.md:96]: Any particular reason for the 1.16 (+downward) incompatibility? Mainly curious, I'm also happy with the answer being "let's just not care for more releases, and encourage people to upgrade".

Comment (lilic) [reply README.md:96]: We do not test that in this branch, that is the only reason, we can't guarantee it, in the other branches we run our e2e tests against those k8s versions so we can say it passes.

At some point we have to decide how many versions we want to support.

**Comment by lilic** [comment]
Want to bump k8s version we test against in master first to 1.18 and bring that commit here.

**Comment by brancz** [comment]
lgtm

**Comment by dgrisonnet** [comment]
You will also need to run the tests on `1.17` like we did here: #476

File: README.md (hunk 1 above)
Comment (brancz) [reply README.md:96]: just wanted to make sure we're aligned, totally ok with this

**Comment by lilic** [comment]
Yep, waiting for master PR to merge first to bring that commit in only figured out we do not test against 1.18 when I was editing the README, hence the WIP :)

**Comment by lilic** [comment]
Decided to merge this first, to unblock CMO stuff, as 1.18 in master is taking a bit longer. @dgrisonnet @brancz this just pins deps PTAL, thanks!

File: jsonnet/kube-prometheus/jsonnetfile.json [hunk 2]
```diff
@@ -72,7 +72,7 @@
           "subdir": "docs/node-mixin"
         }
       },
-      "version": "master"
+      "version": "v0.18.1"
```
Comment (dgrisonnet) [inline jsonnet/kube-prometheus/jsonnetfile.json:75]: Is there a reason why we are using a tag instead of the branch `release-0.18` here?

Comment (lilic) [reply jsonnet/kube-prometheus/jsonnetfile.json:75]: actually noticed that this does not work at all, but somehow our CI did not catch this. Before with make generate-in-docker we ran jb update/install now we skip this step so when running make generate it did not catch this, I am already fixing this.

Comment (dgrisonnet) [reply jsonnet/kube-prometheus/jsonnetfile.json:75]: I looked into it and it appears that we never ran `jb update` in the CI: https://travis-ci.org/github/coreos/kube-prometheus/jobs/674766820#L261.
It is not linked to moving away from `generate-in-docker` but to the lack of `jb update`

Comment (lilic) [reply jsonnet/kube-prometheus/jsonnetfile.json:75]: yes, true! My mistake, this should fix it https://github.com/coreos/kube-prometheus/pull/502

**Pull request merged**